miden-crypto/src/merkle/mmr/partial.rs

```diff
@@ -348,18 +348,29 @@ impl PartialMmr {
 
     /// Removes a leaf of the [PartialMmr] and the unused nodes from the authentication path.
     ///
+    /// Returns a vector of the authentication nodes removed from this [PartialMmr] as a result
+    /// of this operation. This is useful for client-side pruning, where the caller needs to know
+    /// which nodes can be deleted from storage.
+    ///
     /// Note: `leaf_pos` corresponds to the position in the MMR and not on an individual tree.
-    pub fn untrack(&mut self, leaf_pos: usize) {
+    pub fn untrack(&mut self, leaf_pos: usize) -> Vec<(InOrderIndex, Word)> {
```

Comment:
JFYI: for the specific client-related feature request, returning only `InOrderIndex` is fine (because we are only interested in identifying what to prune from the database of nodes we already know). Consistency-wise though it's probably fine to return the hash as well.